src/main/java/Event.java

```diff
@@ -0,0 +1,34 @@
+public class Event extends Task {
+
+    protected String start;
+    protected String end;
+    public Event(String description) {
+        super(description);
+        String[] lineComponents = description.split("/", 2);
+        String[] descriptionComponents = lineComponents[0].split(" ",2);
+        String[] periodComponents = lineComponents[1].split("/",2);
+
+        /** Array containing start of period and prepositions */
+        String[] startPrepositions = periodComponents[0].split(" ", 2);
+
+        /** Array containing end of period and prepositions */
+        String[] endPrepositions = periodComponents[1].split(" ", 2);
+
+        this.start = "(" + startPrepositions[0] + ": " + startPrepositions[1];
+        this.end = endPrepositions[0] + ": " + endPrepositions[1] +")";
+        this.description = descriptionComponents[1] + this.start + this.end;
+        this.isDone = false;
+        this.taskLabel = "[E]";
```

Comment:
consider using regex to split the string easier `splitIntoArgs[1].split(" \\/(from|to) ");`

Reply:
Hi Zhen Yang, appreciate your feedback. What should this code replace?